app/internal/random/random.go

```diff
@@ -4,7 +4,13 @@ import (
 	"crypto/rand"
 )
 
-const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+const charset = "abcdefghijklmnpqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ123456789"
+
+type RandomGenerator interface {
+	String(length int) string
+}
```

Comment:
So we can mock out randomness

Reply:
Unnecessary since you can have functions take a `func(int) string` argument and pass in `random.String`. Take a look at `page/login.go` and how it is injected in `main.go`

Reply:
Nice, didn't realise this was a thing